app/controllers/goals_controller.rb

```diff
@@ -0,0 +1,98 @@
+class GoalsController < ApplicationController
+
+  def index
+    @messages = Message.broadcast
+    @sorting_method = params[:sort]
+    if @sorting_method == "Newest"
+      @goals = Goal.sort_time
+    elsif @sorting_method == "Submitted"
+      @goals = Goal.sort_setter
+    elsif @sorting_method == "Charity"
+      @goals = Goal.sort_charity
+    end
```

Comment:
You should add an else here, otherwise you don't load any goals at all if the parameter is missing